### ISSUE TITLE
google.api_core.exceptions.DeadlineExceeded: 504 Deadline Exceeded 

### ISSUE THREAD

**Issue by Asrst** [google-gemini/generative-ai-python#294]
### Description of the bug:

Request times out, its client side error.

### Actual vs expected behavior:

model: "gemini-1.5-pro-latest"
`response = model.generate_content(large_prompt)`
Wait & Return the response.

**Comment by Asrst** [comment]
sloved with `model.generate_content(prompt, request_options={"timeout": 600})`

**Comment by gcsiapn** [comment]
> sloved with `model.generate_content(prompt, request_options={"timeout": 600})`

I tried it and not work.

**Comment by anakette** [comment]
Having the same issue here, also with 1.5

**Comment by hemanthvatsava** [comment]
Tried it ended with Error: 504 Deadline Exceeded

**Comment by AR-BABER** [comment]
is there a way to request timeout for using startchat instead of generate content?

**Comment by SohamTolwala** [comment]
any solutions to this? I am getting the same error but while embedding my text with an embedding model

here is an overview of my problem and what all I have tried:

`GoogleGenerativeAIError: Error embedding content: 504 Deadline Exceeded`
my code snippet:

def create_subject_vector_db(sub_name):

    chunks_docs = load_and_chunk(f"/content/{sub_name}")

    CHROMA_DB_DIRECTORY = f'/content/CHROMA_DB_DIRECTORY_{sub_name}'

 

    embedding = GoogleGenerativeAIEmbeddings(model="models/embedding-001", google_api_key=google_api_key, request_options={'retry':retry.Retry()})

    vector_db = Chroma.from_documents(documents=chunks_docs, embedding=embedding, persist_directory=CHROMA_DB_DIRECTORY)

    vector_db.persist()

    retriever = vector_db.as_retriever()

    

    return retriever
I have tried splitting the load i.e. chunks into more smaller batches, tried the retry method with Exponential Backoff, also tried it running from another deivce (assuming it is a network error)

**Comment by marcioselemen** [comment]
Same issue here.

**Comment by diegoalejandrogomezmarquitoiii** [comment]
+1

**Comment by mselemen** [comment]
Try to downgrade the following libraries. I am using these and worked:
!pip install -q google-generativeai==0.6.0
!pip install -q langchain-google-genai==1.0.3

**Comment by jacobwu25** [comment]
Getting 504 Deadline Exceeded error with simple one sentence text embedding. Code as below. I'm sure it's not the too long. Anyone has idea?
embeddings = GoogleGenerativeAIEmbeddings(model="text-embedding-004", google_api_key = google_api_key)
embeddings.embed_query("What's our Q1 revenue?")

**Comment by Sonali-Behera-TRT** [comment]
> Try to downgrade the following libraries. I am using these and worked: !pip install -q google-generativeai==0.6.0 !pip install -q langchain-google-genai==1.0.3

Yes, with `langchain-google-genai` version `1.0.3`, it's working now. Thanks for sharing!

**Comment by Travis-Barton** [comment]
+1
